packages/data-context/src/sources/VersionsDataSource.ts

```diff
@@ -100,8 +100,9 @@ export class VersionsDataSource {
   }
 
   private async getVersionMetadata (): Promise<Record<string, string>> {
```

Comment:
kinda a general question - shouldn't an error here basically never crash the process? at worst, we should be blanking the GUI and displaying the error in open mode. why can this even propagate out? is there work formalized to fix this?

we'll certainly have other exceptions in this layer, none of which should probably fully crash Cy in open mode

Reply:
It's b/c there wasn't a catch handler here: https://github.com/cypress-io/cypress/blob/ce647c29df7afc230e1189e3ec728787b1dd16b1/packages/data-context/src/sources/VersionsDataSource.ts#L94-L98

It's not a common pattern for us to do that but we were doing it here b/c we weren't wanting to block the caller on awaiting it

Reply:
But what I'm saying is, if the `this.ctx.update` callback throws, that still shouldn't crash open mode. Cypress is a GUI app, we should recover as gracefully as we can in open mode. Of course things like actual unhandled exceptions/rejections should crash the process, but for things like this, there *is* an appropriate point in the call stack where we could give a better error/recovery steps to the error?

My point is that it shouldn't be possible for GUI to crash unless something is irrecoverably wrong. That's kinda the point desktop-gui was at.

Reply:
Yeah, agreed we should figure out better ways to do a better job to surface the issue and allow to continue when in open mode.